java/compaction-core/src/main/java/sleeper/compaction/job/status/CompactionJobStatusesBuilder.java

```diff
@@ -15,60 +15,94 @@
  */
 package sleeper.compaction.job.status;
 
-import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.TreeMap;
+import java.util.Optional;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class CompactionJobStatusesBuilder {
+    private final Map<String, CompactionJobStatusUpdateRecord> createdUpdateByJobId = new HashMap<>();
+    private final Map<String, List<CompactionJobStatusUpdateRecord>> runUpdatesByJobId = new HashMap<>();
 
-    private final Map<String, CompactionJobCreatedStatus> createdById = new TreeMap<>(); // Order by job ID for output
-    private final Map<String, CompactionJobStartedStatus> startedById = new HashMap<>();
-    private final Map<String, CompactionJobFinishedStatus> finishedById = new HashMap<>();
-    private final Map<String, Instant> expiryDateById = new HashMap<>();
-
-    public CompactionJobStatusesBuilder jobCreated(
-            String jobId, CompactionJobCreatedStatus statusUpdate) {
-        createdById.put(jobId, statusUpdate);
-        return this;
-    }
-
-    public CompactionJobStatusesBuilder jobStarted(
-            String jobId, CompactionJobStartedStatus statusUpdate) {
-        startedById.put(jobId, statusUpdate);
+    public CompactionJobStatusesBuilder jobUpdates(List<CompactionJobStatusUpdateRecord> jobUpdates) {
+        jobUpdates.forEach(this::jobUpdate);
         return this;
     }
 
-    public CompactionJobStatusesBuilder jobFinished(
-            String jobId, CompactionJobFinishedStatus statusUpdate) {
-        finishedById.put(jobId, statusUpdate);
+    public CompactionJobStatusesBuilder jobUpdate(CompactionJobStatusUpdateRecord jobUpdate) {
+        if (jobUpdate.getStatusUpdate() instanceof CompactionJobCreatedStatus) {
+            createdUpdateByJobId.put(jobUpdate.getJobId(), jobUpdate);
+        } else {
+            runUpdatesByJobId.computeIfAbsent(jobUpdate.getJobId(), id -> new ArrayList<>())
+                    .add(jobUpdate);
+        }
         return this;
     }
 
     public Stream<CompactionJobStatus> stream() {
-        return createdById.entrySet().stream()
+        return createdUpdateByJobId.entrySet().stream()
+                .sorted(Comparator.comparing(update -> update.getValue().getStatusUpdate().getUpdateTime()))
                 .map(entry -> fullStatus(entry.getKey(), entry.getValue()));
```

Comment:
With the new sorting here, we might be able to remove some of the sorting in the DynamoDB code?